geth/node/manager.go

```diff
@@ -70,13 +76,91 @@ func (m *NodeManager) startNode(config *params.NodeConfig) (<-chan struct{}, err
 		return nil, ErrNodeExists
 	}
 
-	ethNode, err := MakeNode(config)
+	m.nodeSyncStarted = make(chan struct{}, 0)
+	m.nodeSyncError = make(chan struct{}, 0)
+	m.nodeSyncCompleted = make(chan struct{}, 0)
+
+	ethNode, err := MakeNode(config, func(leth *les.LightEthereum, stack *node.Node, config *params.NodeConfig) {
+		downlder := leth.Downloader()
+
+		go func() {
```

Comment:
Well it would be more sensible to move it into `Node` method, but we will need to make that in our fork of `go-ethereum`. Has to encapsulating into MakeNode, then we must ensure to pass channels around, which would bloat up the argument list. I find this rather too ugly.

The method is ran in https://github.com/status-im/status-go/pull/271/files#diff-32d99350ef62de1b6ea93cc22f7d66aaR182, unfortunately this will block, due to the need to poll for status change, hence the need for the go-routine.

Reply:
Thanks for explanation. One more idea :D Would it be possible to retrieve `les.LightEthereum` service after `MakeNode()` returns and then run `leth.Downloader()` etc?